src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/ListViewRenderer.cs

```diff
@@ -578,18 +578,42 @@ async void OnViewChangeCompleted(object sender, SemanticZoomViewChangedEventArgs
 			ScrollTo(listProxy.ProxiedEnumerable, listProxy[0], ScrollToPosition.Start, true, true);
 		}
 
-		bool ScrollToItemWithAnimation(ScrollViewer viewer, object item)
+		bool ScrollToItemWithAnimation(ScrollViewer viewer, object item, ScrollToPosition toPosition)
 		{
 			var selectorItem = List.ContainerFromItem(item) as Microsoft.UI.Xaml.Controls.Primitives.SelectorItem;
 			var transform = selectorItem?.TransformToVisual(viewer.Content as UIElement);
 			var position = transform?.TransformPoint(new global::Windows.Foundation.Point(0, 0));
 			if (!position.HasValue)
 				return false;
 			// scroll with animation
-			viewer.ChangeView(position.Value.X, position.Value.Y, null);
+			switch (toPosition)
+			{
+				case ScrollToPosition.Center:
+				case ScrollToPosition.End:
+					{
+						double offset = viewer.ViewportHeight - GetContentDesiredHeight(viewer, item);
```

Comment:
Does `selectorItem` not have the size that can be used? I know we create a new/temporary item in the other scenario below, but I was wondering if it is really needed here? `SelectorItem` seems to be a `ContentControl` which should have a size?

Reply:
@mattleibow, I referred to the item size calculation logic from the new item scenarios below ([Referred-Logics](https://github.com/dotnet/maui/blob/29f3c7a72945b9ee8f6f5d1acc9ab4535f1c33c4/src/Controls/src/Core/Compatibility/Handlers/ListView/Windows/ListViewRenderer.cs#L648)). As suggested, by checking the selectorItem size instead of the item size, the logic works correctly for various ScrollPosition scenarios. We updated the logic based on the selectorItem. Could you please review it?